routes/api_v1_0.py

```diff
@@ -1498,9 +1499,9 @@ def _cache_and_send_img(bytesIOBuff: BytesIO, f: str):
     bytesIOBuff: BytesIO object containing image data
     f: filename of image to be cached
     """
-    p = os.path.dirname(f)
-    if not os.path.isdir(p):
-        os.makedirs(p)
+    p = PurePath(f).parent
+    if not Path(p).is_dir():
+        Path(p).mkdir(parents=True, exist_ok=True)
```

Comment:
It's better to use `Path()` instead of `PurePath()` in places like this. Anytime you need to do an IO operation like `mkdir` you can just create one `Path()` object and use it as needed. 

Something like this would probably work here:
``` suggestion 
    p = Path(f).parent
    if not p.is_dir():
        p.mkdir(parents=True, exist_ok=True)
```